### PR TITLE
US477041: Auto-generate a .NET client

### PR DESCRIPTION
**Pull request by kshirsagarnaresh**
https://internal.almoctane.com/ui/entity-navigation?p=131002/6001&entityType=work_item&id=477041

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -52,8 +52,21 @@
         </developer>
     </developers>
 
+    <organization>
```
Comment (dermot-hardy) [inline pom.xml:55]: These should be inherited from the parent.

File: adapter-rest-dotnet-client/pom.xml [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+                            <goal>exec</goal>
+                        </goals>
+                        <configuration>
+                            <executable>dotnet</executable>
+                            <arguments>
+                                <argument>build</argument>
+                            </arguments>
+                        </configuration>
+                    </execution>
+                    <!-- <execution>
```
Comment (dermot-hardy) [inline adapter-rest-dotnet-client/pom.xml:117]: Remove the test section instead of commenting it out.

Comment (kshirsagarnaresh) [reply adapter-rest-dotnet-client/pom.xml]: Removed

**Comment by kshirsagarnaresh** [comment]
> As discussed, the copyright etc. is not being set in the auto-generated code.

Hi @dermot-hardy,
Checked for 'Copyright', 'Author' and 'Company Name' attribute, actually there is no attribute present for `generatorName = [csharp-netcore](https://openapi-generator.tech/docs/generators/csharp-netcore/)`, but same things are present with `generatorName = [aspnetcore](https://openapi-generator.tech/docs/generators/aspnetcore/)`. 

Even thogh, I added attribute between line no. [#71 - #74](https://github.com/FileAnalysisSuite/adapter-rest-contract/pull/5/files#diff-5139d1da87c038f5b28cb2f74983845d95d1844d4f2bf08a9652e37524bf44f8R71) of `aspnetcore` for `csharp-netcore`.

**Comment by dermot-hardy** [comment]
If there is nothing like that then introduce a script to modify the auto-generated project file before it is compiled.

See [`caf-storage-service-client/pom.xml:124`](https://github.houston.softwaregrp.net/caf/caf-storage/blob/v6.6.0-source/caf-storage-service-client/pom.xml#L124) as an example of a plugin being used to adjust auto-generated code.

**Comment by kshirsagarnaresh** [comment]
> That's unfortunate. Are there extra arguments that could be added to the `dotnet` commands to allow you to set the properties even though they're not in the project file? So where at the minute we do `dotnet build` could we instead do `dotnet build -p:Copyright=MicroFocus`? Is there a syntax that would allow us to do something like that?

Tried with this `dotnet build -p:Copyright=MicroFocus`, this also not replacing value.

**Comment by kshirsagarnaresh** [comment]
> If there is nothing like that then introduce a script to modify the auto-generated project file before it is compiled.
> 
> See [`caf-storage-service-client/pom.xml:124`](https://github.houston.softwaregrp.net/caf/caf-storage/blob/v6.6.0-source/caf-storage-service-client/pom.xml#L124) as an example of a plugin being used to adjust auto-generated code.

Yes, tried with groovy script and able to replace attributes from .csproj file.

File: adapter-rest-dotnet-client/pom.xml [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+                                </script>
+                            </scripts>
+                        </configuration>
+                    </execution>
+                </executions>
+                <dependencies>
+                    <dependency>
+                        <groupId>org.codehaus.groovy</groupId>
+                        <artifactId>groovy-all</artifactId>
+                        <version>2.4.7</version>
```
Comment (dermot-hardy) [inline adapter-rest-dotnet-client/pom.xml:130]: Let's update to a newer version.

File: adapter-rest-dotnet-client/pom.xml [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+                                    if(fileToModifyTextAssets == null){
+                                            throw Exception();
+                                    }
+                                    def modifiedAuthorText = fileToModifyTextAssets.replaceAll('<Authors>OpenAPI</Authors>', '<Authors>Krzysztof Ploch</Authors>')                                    
+
+                                    def modifiedCopyrightText = modifiedAuthorText.replaceAll('<Copyright>No Copyright</Copyright>', '<Copyright>${copyrightNotice}</Copyright>')                                    
+
+                                    def modifiedAssemblyTitleText = modifiedCopyrightText.replaceAll('<AssemblyTitle>OpenAPI Library</AssemblyTitle>', '<AssemblyTitle>REST Adapter Client Library</AssemblyTitle>')                                    
+
+                                    def modifiedDescriptionText = modifiedAssemblyTitleText.replaceAll('<Description>A library generated from a OpenAPI doc</Description>', '<Description>REST Adapter and auto code generator</Description>')
```
Comment (dermot-hardy) [inline adapter-rest-dotnet-client/pom.xml:116]: This should be a description of the auto-generated client.

File: adapter-rest-dotnet-client/pom.xml [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+                                    }
+                                    
+                                    def fileToModifyAssets = new File(targetDir + "/openapi-codegen/src/MicroFocus.FAS.Adapters.Rest.Client/MicroFocus.FAS.Adapters.Rest.Client.csproj")
+                                    def fileToModifyTextAssets = fileToModifyAssets.text
+                                    if(fileToModifyTextAssets == null){
+                                            throw Exception();
+                                    }
+                                    def modifiedAuthorText = fileToModifyTextAssets.replaceAll('<Authors>OpenAPI</Authors>', '<Authors>Krzysztof Ploch</Authors>')                                    
+
+                                    def modifiedCopyrightText = modifiedAuthorText.replaceAll('<Copyright>No Copyright</Copyright>', '<Copyright>${copyrightNotice}</Copyright>')
```
Comment (dermot-hardy) [inline adapter-rest-dotnet-client/pom.xml:112]: This should be passed in as a parameter, and it should be being escaped.

**Review by dermot-hardy — CHANGES_REQUESTED** [review]

File: adapter-rest-dotnet-client/pom.xml [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+                                <property>
+                                    <name>targetDir</name>
+                                    <value>${project.build.directory}</value>
+                                </property>
+                            </properties>
+                            <scripts>
+                                <script><![CDATA[
+                                    new AntBuilder().copy( todir:targetDir + '/openapi-codegen/src/MicroFocus.FAS.Adapters.Rest.Client' ) {
+                                        fileset( dir:targetDir + '/openapi-codegen/src/MicroFocus.FAS.Adapters.Rest.Client' )
+                                    }
```
[the remark this answers is not in the document]
Comment (dermot-hardy) [reply adapter-rest-dotnet-client/pom.xml]: Removed in 73e01c4.

File: adapter-rest-dotnet-client/pom.xml (hunk 3 above)
Comment (dermot-hardy) [reply adapter-rest-dotnet-client/pom.xml]: Updated in 73e01c4.

File: adapter-rest-dotnet-client/pom.xml (hunk 4 above)
Comment (dermot-hardy) [reply adapter-rest-dotnet-client/pom.xml]: Updated in 73e01c4.

File: adapter-rest-dotnet-client/pom.xml (hunk 5 above)
Comment (dermot-hardy) [reply adapter-rest-dotnet-client/pom.xml]: Updated in 73e01c4.

**Review by dermot-hardy — APPROVED** [review]

**Pull request merged**